tcms/testcases/tests/test_views.py

```diff
@@ -179,6 +179,21 @@ def setUpTestData(cls):
         user_should_have_perm(cls.tester, 'testcases.add_testcase')
         user_should_have_perm(cls.tester, 'testcases.view_testcase')
 
+    def test_form_invalid(self):
+
+        data = self.data.copy()
+        error_messages = ["Ensure this value has at most 255 characters",
+                          "Unknown user:", "This field is required."]
+        data['summary'] = 'Lorem'*54
+        data['default_tester'] = 'NotARealUser'
+        data['product'] = ''
+        data['category'] = ''
+
+        response = self.client.post(self.new_case_url, data)
+
+        for error_message in error_messages:
+            self.assertContains(response, error_message)
+
```

Comment:
This scenario is covering the case where the main form is invalid, which isn't handled directly by our implementation so we don't need this. For completeness this is handled by Django's CreateView (FormView in reality) and we trust that Django is doing the right thing here in terms of displaying errors.

However we're interested in the inline form called notify_formset in the implementation. There's no test to validate whether or not we're handling the case when this becomes invalid. See testplans/tests/test_views.py around line 70 for clues.

Reply:
@atodorov I have made the changes. Please let me know if they are okay.